internal/archive/archive.go

```diff
@@ -148,6 +195,18 @@ func openUbuntu(options *Options) (Archive, error) {
 		pubKeys: options.PubKeys,
 	}
 
+	if archive.baseURL == "" {
+		archive.baseURL = archiveURL(options.Pro, options.Arch)
+	}
```

Comment:
`baseURL` can never be set if we are creating the archive just above this block, right? Can you instead call `archiveURL` when defining the archive struct?

Reply:
Changed.